src/sidebar/oauth-auth.js

```diff
@@ -16,23 +16,62 @@ function auth($http, settings) {
   var accessTokenPromise;
   var tokenUrl = resolve('token', settings.apiUrl);
 
+  // Return a new tokenInfo object from the given tokenUrl endpoint response.
+  function tokenInfoFrom(response) {
+    var data = response.data;
+    return {
+      accessToken:  data.access_token,
+      expiresIn:    data.expires_in,
```

Comment:
Hmm. `expires_in` is part of the OAuth spec. I did change it to `javaScript` case styling but I think maybe we should stick to the spec's names for things as much as possible? So I think adding a typedef sounds good.